File: binding/src/libqgenericpluginfactory.cpp
#include <QGenericPluginFactory>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qgenericpluginfactory.h>
#include "libqgenericpluginfactory.h"
#include "libqgenericpluginfactory.hxx"

QGenericPluginFactory* QGenericPluginFactory_new(const QGenericPluginFactory* other) {
	 return new QGenericPluginFactory(*other);
}

QGenericPluginFactory* QGenericPluginFactory_new2(QGenericPluginFactory* other) {
	 return new QGenericPluginFactory(*other);
}

QGenericPluginFactory* QGenericPluginFactory_new3() {
	 return new QGenericPluginFactory();
}

QGenericPluginFactory* QGenericPluginFactory_new4(const QGenericPluginFactory* param1) {
	 return new QGenericPluginFactory(*param1);
}

void QGenericPluginFactory_CopyAssign(QGenericPluginFactory* self, QGenericPluginFactory* other) {
    *self = *other;
}

void QGenericPluginFactory_MoveAssign(QGenericPluginFactory* self, QGenericPluginFactory* other) {
    *self = std::move(*other);
}

libqt_list QGenericPluginFactory_Keys() {
	auto _ret = QGenericPluginFactory::keys();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		QByteArray _b = _ret[_i].toUtf8();
		libqt_string* _str = new libqt_string();
		_str->len = _b.length();
		_str->data = static_cast<const char*>(malloc(_str->len + 1));
		memcpy((void*)_str->data, _b.data(), _str->len);
		((char*)_str->data)[_str->len] = '\0';
		_data[_i] = _str;
	}
	return _arr;
}

QObject* QGenericPluginFactory_Create(const libqt_string param1, const libqt_string param2) {
	return QGenericPluginFactory::create(QString::fromUtf8(param1.data, param1.len), QString::fromUtf8(param2.data, param2.len));
}

void QGenericPluginFactory_OperatorAssign(QGenericPluginFactory* self, const QGenericPluginFactory* param1) {
	self->operator=(*param1);
}

void QGenericPluginFactory_Delete(QGenericPluginFactory* self) {
    delete self;
}

File: binding/src/libqgenericpluginfactory.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgenericpluginfactory.html
pub const qgenericpluginfactory = struct {

    /// New constructs a new QGenericPluginFactory object.
    pub fn New(other: ?*anyopaque) QtC.QGenericPluginFactory {
        return qtc.QGenericPluginFactory_new(@ptrCast(other));
    }


    /// New2 constructs a new QGenericPluginFactory object.
    pub fn New2(other: ?*anyopaque) QtC.QGenericPluginFactory {
        return qtc.QGenericPluginFactory_new2(@ptrCast(other));
    }


    /// New3 constructs a new QGenericPluginFactory object.
    pub fn New3() QtC.QGenericPluginFactory {
        return qtc.QGenericPluginFactory_new3();
    }


    /// New4 constructs a new QGenericPluginFactory object.
    pub fn New4(param1: ?*anyopaque) QtC.QGenericPluginFactory {
        return qtc.QGenericPluginFactory_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QGenericPluginFactory_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QGenericPluginFactory_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Keys() []const u8 {
        return qtc.QGenericPluginFactory_Keys();
    }

    pub fn Create(param1: []const u8, param2: []const u8) QtC.QObject {
        const param1_str = qtc.libqt_string{
    .len = param1.len,
    .data = param1.ptr,
};
const param2_str = qtc.libqt_string{
    .len = param2.len,
    .data = param2.ptr,
};
return qtc.QGenericPluginFactory_Create(param1_str, param2_str);
    }

    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QGenericPluginFactory_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGenericPluginFactory_Delete(@ptrCast(self));
    }
};
